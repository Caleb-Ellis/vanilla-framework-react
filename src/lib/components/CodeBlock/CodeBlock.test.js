import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import CodeBlock from './CodeBlock';

describe('CodeBlock component', () => {
  it('should render one line of code correctly', () => {
    const codeBlock = ReactTestRenderer.create(
      <CodeBlock>
        {'this is a single line of code'}
      </CodeBlock>,
    );
    const json = codeBlock.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render multiple lines of code correctly', () => {
    const codeBlock = ReactTestRenderer.create(
      <CodeBlock>
        {`this is code sample line 1
          this is code sample line 2
          this is code sample line 3`}
      </CodeBlock>,
    );
    const json = codeBlock.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render one line of code correctly with numbered modifier', () => {
    const codeBlock = ReactTestRenderer.create(
      <CodeBlock modifier="numbered">
        {'this is a single line of code'}
      </CodeBlock>,
    );
    const json = codeBlock.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render multiple lines of code correctly with numbered modifier', () => {
    const codeBlock = ReactTestRenderer.create(
      <CodeBlock modifier="numbered">
        {`this is code sample line 1
          this is code sample line 2
          this is code sample line 3`}
      </CodeBlock>,
    );
    const json = codeBlock.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should display a single blank line if component is empty', () => {
    const codeBlock = ReactTestRenderer.create(
      <CodeBlock />,
    );
    const json = codeBlock.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render markup correctly', () => {
    const codeBlock = ReactTestRenderer.create(
      <CodeBlock>
        <head>
          <title>Markup Example</title>
        </head>
        <body>
          <div>This is an example of using markup in the CodeBlock component</div>
          <div>This & should escape</div>
        </body>
      </CodeBlock>,
    );
    const json = codeBlock.toJSON();
    expect(json).toMatchSnapshot();
  });
});
