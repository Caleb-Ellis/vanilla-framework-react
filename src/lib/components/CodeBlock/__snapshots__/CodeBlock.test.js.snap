// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock component should display a single blank line if component is empty 1`] = `
<pre
  className={null}
>
  <code>
    
  </code>
</pre>
`;

exports[`CodeBlock component should render markup correctly 1`] = `
<pre
  className={null}
>
  <code>
    &lt;head&gt;
  &lt;title&gt;Markup Example&lt;/title&gt;
&lt;/head&gt;
&lt;body&gt;
  &lt;div&gt;This is an example of using markup in the CodeBlock component&lt;/div&gt;
  &lt;div&gt;This &amp; should escape&lt;/div&gt;
&lt;/body&gt;
  </code>
</pre>
`;

exports[`CodeBlock component should render multiple lines of code correctly 1`] = `
<pre
  className={null}
>
  <code>
    this is code sample line 1
this is code sample line 2
this is code sample line 3
  </code>
</pre>
`;

exports[`CodeBlock component should render multiple lines of code correctly with numbered modifier 1`] = `
<pre
  className="p-code-numbered"
>
  <code>
    <span
      className="code-line"
    >
      this is code sample line 1
    </span>
    <span
      className="code-line"
    >
      this is code sample line 2
    </span>
    <span
      className="code-line"
    >
      this is code sample line 3
    </span>
  </code>
</pre>
`;

exports[`CodeBlock component should render one line of code correctly 1`] = `
<pre
  className={null}
>
  <code>
    this is a single line of code
  </code>
</pre>
`;

exports[`CodeBlock component should render one line of code correctly with numbered modifier 1`] = `
<pre
  className="p-code-numbered"
>
  <code>
    <span
      className="code-line"
    >
      this is a single line of code
    </span>
  </code>
</pre>
`;
